src/SFML/Audio/InputSoundFile.cpp

```diff
@@ -195,15 +196,42 @@ unsigned int InputSoundFile::getSampleRate() const
 ////////////////////////////////////////////////////////////
 Time InputSoundFile::getDuration() const
 {
-    return seconds(static_cast<float>(m_sampleCount) / m_channelCount / m_sampleRate);
+    // Make sure we don't divide by 0
+    if (m_channelCount != 0 && m_sampleRate != 0)
+        return seconds(static_cast<float>(m_sampleCount) / m_channelCount / m_sampleRate);
+    else
+        return Time::Zero;
+}
```

Comment:
Not sure if the rest of the team agrees, but I feel one if statement could work nicely here:

```cpp
     // Make sure we don't divide by 0
    if (m_channelCount == 0 || m_sampleRate == 0)
        return Time::Zero;

    return seconds(static_cast<float>(m_sampleCount) / m_channelCount / m_sampleRate);
```

Same goes for `getTimeOffset` below.